server/routes/applications.js

```diff
@@ -10,25 +10,38 @@ const validator = new Validator();
 const routeUtils = require(`${serverRoot}/routes/utils`);
 const routeViews = 'applications';
 
-router.get('(/manage-applications)?', (req, res, next) => {
-  logger.info(`GET request to serve index page: ${req.path}`);
-  const viewData = {
+function createViewData (title, activePage) {
+  return {
     this_data: null,
     this_errors: null,
-    active_page: 'application-overview',
-    title: 'Application overview'
+    active_page: title,
+    title: activePage,
+    future_flag: process.env.APPLICATIONS_DEVELOPER_SERVICE_DISPLAY_FUTURE_FLAG
   };
-  Promise.all(
-    [
-      applicationsDeveloperService.getApplicationList('live'),
-      applicationsDeveloperService.getApplicationList('test'),
-      applicationsDeveloperService.getApplicationList('future')
-    ]
-  ).then(([listLive, listTest, listFuture]) => {
-    viewData.this_data = {
-      live: listLive.data,
-      test: listTest.data,
-      future: listFuture.data
+}
+
+router.get('(/manage-applications)?', (req, res, next) => {
+  logger.info(`GET request to serve index page: ${req.path}`);
+  const viewData = createViewData('Application overview', 'application-overview');
+  const applicationQueries = [
+    applicationsDeveloperService.getApplicationList('live'),
+    applicationsDeveloperService.getApplicationList('test')
+  ];
+  if (viewData.future_flag === 'true') {
+    applicationQueries.push(applicationsDeveloperService.getApplicationList('future'));
+  };
+  Promise.all(applicationQueries).then(([listLive, listTest, listFuture]) => {
```

Comment:
I don't know quite what you're asking.
That we should always query future regardless of the future flag and let logic in the view sort it out?
Or that we can statically reference the flag in the view?
Or that we should include the view as a `const future_flag = require(...)`

Reply:
basically, revert to previous version of this method. only do the toggle thing in the view

Reply:
And you're sure that will not error in live where there isn't a future environment to query?

Reply:
you can export the toggle value from `app.js` by adding it to nunjucks globals. nothing to do in the route handler, not for MVP at least..

Reply:
The route handler will error if it makes a query to a url that doesn't exist. Do you want me to swap onto `Promise.allSettled()` rather than `Promise.all`

Reply:
I prefered to use the logic as it meant that I didn't have to rework the error handling logic.

Reply:
Globals changed. However based on discussion with team they are satisfied with this logic and hasn't been changed.